### ISSUE TITLE
InvalidProfileResponse with Github

### ISSUE THREAD

**Issue by adarqui** [freizl/hoauth2#49]
Hey,

Just wondering if you have any idea why i'm receiving this error when trying to authenticate with Github. I'm using yesod-auth-oauth2, which uses hoauth2. This worked several months ago. I've just recently updated all of my deps to the latest. I will try to figure it out myself but, just wondering if you've ever seen this.

```
InvalidProfileResponse "github" "hoauth2.HttpClient.parseResponseJSON/Could not decode JSON:
```

edit: I have a feeling it has to do with some fields being "null" which were never null before, ie, name & email.

Thanks!

**Comment by freizl** [comment]
would you mind put more details here just in case people may google same issue as yours' but ended up here without any clues..

**Comment by adarqui** [comment]
Hey! Sure.

So I actually figured it out last night. Turns out yesod-auth-oauth2 added support for the "location" field in the github oauth2 response. However this field can be "null" - and yesod-auth-oauth2 doesn't handle it as a Maybe value. So as of 0.1.9, it yesod-auth-oauth2 basically breaks if you've never specified a Location in your profile/settings.

I have issued a pull request to fix the problem: https://github.com/thoughtbot/yesod-auth-oauth2/pull/64

That fixed it for me. So, hopefully everything is ok with them and it fixes this issue.

It has nothing to do with hoauth2 .. hoauth2 is simply provides the core-routines to communicate with github, via authGetJSON. It's the handling on the yesod-auth-oauth2 side which results causes the error.

Hope that clears it up!

Thanks alot,

-- Andrew
